index.js

```diff
@@ -71,6 +71,16 @@ module.exports = Object.assign(
         }
       },
     }),
+    onFirstPause: buildOperation({
+      name: "onFirstPause",
+      arity: 2,
+      sync: function([item]) {
+        return evaluateSync(item);
+      },
+      async: function([item, cb], resolve, reject) {
+        evaluateAsync(item, resolve, reject, cb);
+      },
+    }),
```

Comment:
I _think_ you can do what you want with something like this?
```
  async: function([item, cb], resolve, reject) {
    let completed = false;

    evaluateAsync(item, value => {
      completed = true;
      resolve(value);
    }, err => {
      completed = true;
      reject(value);
    });

    if (!completed) {
      cb();
    }
  },
```
or alternatively using the existing public API:
```
const runGenerator = gensync(function* (item) {
  return yield* item;
});

const onFirstPause = gensync({
  name: "onFirstPause",
  arity: 2,
  sync: function([item]) {
    return runGenerator.sync(item);
  },
  errback: function([item, firstPause], cb) {
    let completed = false;

    // Gensync calls `errback` callbacks synchronously if it can, even in the async case, if I remember right
    runGenerator.errback(item, (err, value) => {
      completed = true;
      cb(value);
    });

    if (!completed) {
      firstPause();
    }
  },
})
```

Reply:
Ok, I will try the second approach :+1: